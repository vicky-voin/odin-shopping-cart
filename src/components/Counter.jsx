import { useState } from "react";
import styles from "./Counter.module.css";

export default function Counter({ initialCount = 0 }) {
  const [count, setCount] = useState(initialCount);

  return (
    <>
      <div>
        <button>-</button>
        <input
          className={styles.inputValue}
          type="number"
          defaultValue={count}
        ></input>
        <button>+</button>
      </div>
    </>
  );
}
