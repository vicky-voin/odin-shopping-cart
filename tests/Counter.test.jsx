import { screen, render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Counter from "../src/components/Counter";

describe("Counter component tests", () => {
  it("renders a counter with the default value 0", () => {
    const { container } = render(<Counter></Counter>);
    expect(container).toMatchSnapshot();

    const input = screen.getByRole("spinbutton");
    expect(input).toHaveAttribute("value", "0");
  });
});
