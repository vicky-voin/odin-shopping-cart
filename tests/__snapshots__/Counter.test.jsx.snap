// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter component tests > renders a counter with the default value 0 1`] = `
<div>
  <div>
    <button>
      +
    </button>
    <input
      type="number"
      value="0"
    />
    <button>
      -
    </button>
  </div>
</div>
`;
